pkgs/development/haskell-modules/configuration-hackage2nix/main.yaml

```diff
@@ -112,6 +112,7 @@ extra-packages:
   - hspec-discover < 2.8                # 2022-04-07: Needed for tasty-hspec 1.1.6
   - hspec-meta < 2.8                    # 2022-12-07: Needed for elmPackages.elm / hspec-discover
   - hspec-golden == 0.1.*               # 2022-04-07: Needed for elm-format
+  - http2 < 3.3                         # 2023-08-24: Needed for twain
```

Comment:
Can you add a comment which twain version needs the override and a link to a pr or issue upstream.

Reply:
From looking at twain's git history, it has depended upon http2 since version 2.0.0.0 and has had the upper bounds of http2 3.x ever since.  There is no upstream issue for migrating it to the current version of http2.
https://github.com/alexmingoia/twain/commit/2ef20f5df66d1cddb42e8b672d92267c556fdcd5#diff-99c6c7a2194a963526fd5d0f57663534b8b754f09e8ff44717397578ee585a9e

Reply:
Normally, in this case, we open an issue (or PR depending on time and motivation) upstream, where we friendly point out, that there is a newer release of the dependency.

Reply:
I just created an issue, as I barely know Haskell, so I could not do the work myself.